File: tasks/implementations/go/tasks/prime_sieve.go
package main

import "fmt"

func RunPrimeSieve(size string, fixturesRoot string) {
    limits := map[string]int{"s": 50000, "m": 125000, "l": 250000}
    n := limits[size]
    sieve := make([]bool, n+1)
    for i := 2; i <= n; i++ { sieve[i] = true }
    for p := 2; p*p <= n; p++ {
        if sieve[p] {
            for i := p * p; i <= n; i += p { sieve[i] = false }
        }
    }
    count := 0
    for _, prime := range sieve { if prime { count++ } }
    fmt.Println(count)
}

File: tasks/implementations/cpp/tasks/prime_sieve.cpp
#include <iostream>
#include <string>
#include <vector>

void run_prime_sieve(const std::string& size, const std::string& fixturesRoot) {
    int n = size == "s" ? 50000 : size == "m" ? 125000 : 250000;
    std::vector<bool> sieve(n + 1, true);
    sieve[0] = sieve[1] = false;
    for (int p = 2; p * p <= n; ++p) {
        if (sieve[p]) {
            for (int i = p * p; i <= n; i += p) sieve[i] = false;
        }
    }
    int count = 0;
    for (bool prime : sieve) if (prime) count++;
    std::cout << count << std::endl;
}
